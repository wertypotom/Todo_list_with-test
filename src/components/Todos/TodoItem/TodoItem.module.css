.container {
  font-size: 25px;
  letter-spacing: 1px;
  display: flex;
  gap: 20px;
  border-bottom: 1px solid var(--secondary-color);

  padding: 12px;
}

.completed {
  color: var(--secondary-color-accent);
  text-decoration: line-through;
}

label {
  user-select: none;
}
