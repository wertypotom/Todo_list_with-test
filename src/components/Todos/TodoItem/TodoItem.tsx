import { Todo } from '@/types/Todo.type';
import styles from './TodoItem.module.css';
import { Checkbox } from '@/UI';

type Props = {
  todo: Todo;
  handleTodoCheck: (todo: Todo) => void;
};

const TodoItem = ({ todo, handleTodoCheck }: Props) => {
  return (
    <div className={styles.container}>
      <Checkbox
        checked={todo.completed}
        onChange={() => handleTodoCheck(todo)}
        id={`${todo.id} - ${todo.title}`}
      />
      <label
        htmlFor={`${todo.id} - ${todo.title}`}
        className={todo.completed ? styles.completed : ''}
      >
        {todo.title}
      </label>
    </div>
  );
};

export default TodoItem;
