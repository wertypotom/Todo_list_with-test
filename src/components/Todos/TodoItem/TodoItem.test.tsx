import { Todo } from '@/types/Todo.type';
import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import TodoItem from './TodoItem';
import styles from './TodoItem.module.css';

vi.mock('./TodoItem.module.css', () => ({
  default: new Proxy(new Object(), {
    get(_, style) {
      return style;
    },
  }),
}));

const todo: Todo = {
  completed: true,
  id: '0',
  title: 'Hello',
};

describe('TodoItem Component', () => {
  it('Should render Todo Item', () => {
    const { container } = render(
      <TodoItem todo={todo} handleTodoCheck={() => {}} />
    );

    expect(container).toMatchSnapshot();
  });

  it('Should render TodoItem with passed todo', () => {
    const { getByText } = render(
      <TodoItem todo={todo} handleTodoCheck={() => {}} />
    );

    const item = getByText(/hello/i);
    expect(item).toBeInTheDocument();
  });

  it('handleTodoCheck should be called', async () => {
    const handleTodoCheck = vi.fn();

    const { getByRole } = render(
      <TodoItem handleTodoCheck={handleTodoCheck} todo={todo} />
    );

    const checkBox = getByRole('checkbox');
    await userEvent.click(checkBox);
    expect(handleTodoCheck).toHaveBeenCalledOnce();
  });

  it('should have completed class if checked', async () => {
    const handleTodoCheck = vi.fn();

    const { getByText } = render(
      <TodoItem handleTodoCheck={handleTodoCheck} todo={todo} />
    );

    const item = getByText(/hello/i);
    expect(item).toBeInTheDocument();
    expect(item).toHaveClass(styles.completed);
  });
});
