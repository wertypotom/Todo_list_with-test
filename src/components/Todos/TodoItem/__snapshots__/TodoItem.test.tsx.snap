// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TodoItem Component > Should render Todo Item 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="_container_3de4f0"
    >
      <input
        checked=""
        id="0 - Hello"
        type="checkbox"
      />
    </div>
    <label
      class="completed"
      for="0 - Hello"
    >
      Hello
    </label>
  </div>
</div>
`;
